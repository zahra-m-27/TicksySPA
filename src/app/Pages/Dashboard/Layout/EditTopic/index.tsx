import React, { useState } from "react";
import Assets from "../../../../Assets";
import styles from "./styles.module.scss";
import SEInput from "../../../../Components/SEInput";
import TickCard from "../../../../Components/TickCard";
import ClassNames from "../../../../Utilities/ClassNames";

export default function EditTopic() {
  const [StartSearch, setStartSearch] = useState<boolean>(false);

  return (
    <div className={styles.container}>
      <TickCard
        title="اعضا"
        icon={Assets.Svgs.People}
        className={styles.member_card}
        contentClassName={styles.member_content}
      >
        {StartSearch ? (
          <div className={styles.search_box}>
            <SEInput
              onChangeText={() => {}}
              inputClassName={styles.search}
              icon={<img src={Assets.Svgs.Search} />}
              className={styles.search_container}
            />
            <img
              src={Assets.Svgs.Cancel}
              className={styles.close_search}
              onClick={() => setStartSearch(false)}
            />
          </div>
        ) : (
          <div
            onClick={() => setStartSearch(true)}
            className={ClassNames(styles.member_container, styles.add_button)}
          >
            <img src={Assets.Svgs.Plus} />
            <p>افزودن عضو</p>
          </div>
        )}
        <div className={styles.member_container}>
          <img src={Assets.Svgs.Minus} />
          <p>alialavi@gmail.com</p>
        </div>
        <div className={styles.member_container}>
          <img src={Assets.Svgs.Minus} />
          <p>alialavi@gmail.com</p>
        </div>
        <div className={styles.member_container}>
          <img src={Assets.Svgs.Minus} />
          <p>alialavi@gmail.com</p>
        </div>
        <div className={styles.member_container}>
          <img src={Assets.Svgs.Minus} />
          <p>alialavi@gmail.com</p>
        </div>
        <div className={styles.member_container}>
          <img src={Assets.Svgs.Minus} />
          <p>alialavi@gmail.com</p>
        </div>
      </TickCard>
      <TickCard
        title="ویرایش تاپيک"
        icon={Assets.Svgs.Topic}
        className={styles.detail_card}
        buttons={[
          {
            label: "ثبت",
            className: styles.enter_button,
          },
        ]}
      >
        <div className={styles.upload}>
          <label htmlFor="picture">
            <img src={Assets.Svgs.Camera} className={styles.picture} />
          </label>
          <input type="file" id="picture" className={styles.upload_image} />
        </div>
        <div className={styles.inputs_container}>
          <div className={styles.middle}>
            <label>:عنوان</label>
            <SEInput
              onChangeText={() => {}}
              innerContainerClassName={styles.input}
            />
            <label>:شناسه</label>
            <SEInput
              onChangeText={() => {}}
              innerContainerClassName={styles.input}
            />
          </div>
          <div className={styles.below}>
            <label>:توضيح</label>
            <SEInput
              minLines={5}
              onChangeText={() => {}}
              innerContainerClassName={styles.input}
            />
          </div>
        </div>
      </TickCard>
    </div>
  );
}
