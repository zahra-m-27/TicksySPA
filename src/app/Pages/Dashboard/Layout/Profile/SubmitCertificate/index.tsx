import API from '../../../../../API';
import moment from 'jalali-moment';
import Assets from '../../../../../Assets';
import styles from './styles.module.scss';
import {useEffect, useState} from 'react';
import {Button, message, Spin} from 'antd';
import UserIdentityDto from '../../../../../API/DTOs/UserIdentityDto';
import {useHistory} from 'react-router-dom';

export default function SubmitCertificate() {
  const history = useHistory();
  const [Loading, setLoading] = useState(false);
  const [Attachment, setAttachment] = useState<File>();
  const [Identity, setIdentity] = useState<UserIdentityDto>();

  const onSelectFile = (file: File) => {
    setAttachment(file);
  };

  useEffect(() => {
    setLoading(true);
    getIdentity();
  }, []);

  const getIdentity = () => {
    API.Users.GetIdentity({
      status: new URLSearchParams(history.location.search).get('status') ?? '1',
    })
      .then((response) => setIdentity(response))
      .finally(() => setLoading(false));
  };

  const onUploadCertificate = () => {
    setLoading(true);
    API.Users.UpdateIdentity({
      identifier_image: Attachment,
    })
      .then(() => {
        getIdentity();
      })
      .catch(() => {
        setLoading(false);
        message.error('مشکلی در ارسال مدرک احراز هویت رخ داده است.');
      });
  };

  const LoadingComponent = (
    <div className={styles.container_report}>
      <div className={styles.top_part}>
        <div className={styles.title}>
          <img
            alt=""
            src={Assets.Images.FingerPrint}
            className={styles.report_title_image}
          />
          احراز هویت
        </div>
      </div>
      <div className={styles.report_body}>
        <Spin size="large" />
      </div>
    </div>
  );

  if (!Identity) {
    return LoadingComponent;
  }

  if (Identity.status === '3') {
    return (
      <div className={styles.container_upload}>
        <div className={styles.top_part}>
          <div className={styles.title}>
            <img
              alt=""
              className={styles.title_image}
              src={Assets.Images.FingerPrint}
            />
            احراز هویت
          </div>
        </div>
        <div className={styles.upload_body}>
          <p className={styles.upload_text}>
            لطفا سند احراز هویت خود را بارگذاری کنید
          </p>

          <label htmlFor="attachment" className={styles.attachment_box}>
            <div className={styles.input_box}>
              <img
                alt=""
                src={Assets.Images.Upload}
                className={styles.upload_icon}
              />
              <p className={styles.file_name}>
                {Attachment?.name ?? 'انتخاب فایل'}
              </p>
            </div>
          </label>
          <input
            type={'file'}
            id={'attachment'}
            className={styles.input_file}
            onChange={(e) => {
              e.target.files && onSelectFile && onSelectFile(e.target.files[0]);
            }}
          />

          <Button
            type={'primary'}
            loading={Loading}
            onClick={onUploadCertificate}
            className={styles.submit_button}>
            ثبت
          </Button>
        </div>
      </div>
    );
  }

  if (Loading) {
    return LoadingComponent;
  }

  return (
    <div className={styles.container_report}>
      <div className={styles.top_part}>
        <div className={styles.title}>
          <img
            alt=""
            src={Assets.Images.FingerPrint}
            className={styles.report_title_image}
          />
          احراز هویت
        </div>
      </div>
      <div className={styles.report_body}>
        {Identity.status === '1' ? (
          <p className={styles.right_side}>
            <span>سند شما ثبت شد</span>
            <span>لطفا منتظر پاسخ ما باشین</span>
            <span>از صبر و شکیبایی شما </span>
            <span>سپاسگزاریم</span>
          </p>
        ) : (
          <p className={styles.right_side}>
            <span>سند شما تایید شد</span>
            <span>اکنون دسترسی ایجاد تاپیک در سامانه</span>
            <span>برای شما وجود دارد</span>
            <span>با سپاس</span>
          </p>
        )}
        <div className={styles.divider} />
        <div className={styles.left_side}>
          <div className={styles.request_data}>
            <div className={styles.request_container}>
              <span className={styles.request}>وضعیت درخواست:</span>
              <div className={styles.request_data_container}>
                {Identity.status === '1' ? 'در حال بررسی' : 'تایید شده'}
              </div>
            </div>
            <div className={styles.request_container}>
              <span className={styles.request}>تاریخ درخواست:</span>
              <div className={styles.request_data_container}>
                {moment
                  .utc(Identity.request_time ?? new Date())
                  .local()
                  .locale('fa')
                  .format('YYYY/MM/D')}
              </div>
            </div>
            {Identity.status === '1' && (
              <div className={styles.request_container}>
                <span className={styles.request}>تاریخ انقضا:</span>
                <div className={styles.request_data_container}>
                  {moment
                    .utc(Identity.request_time ?? new Date())
                    .add(1, 'week')
                    .local()
                    .locale('fa')
                    .format('YYYY/MM/D')}
                </div>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
}
