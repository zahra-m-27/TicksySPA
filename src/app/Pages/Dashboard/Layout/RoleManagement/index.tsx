import {Button} from 'antd';
import Assets from '../../../../Assets';
import styles from './styles.module.scss';
import showDialog from '../../../../Components/TDialog';
import AddRoleDialog from '../../../../Dialogs/AddRole';
import {useRef} from 'react';

interface RoleData {
  role: string;
  id: number;
}

export default function RoleManagement() {
  const dismissDialog = useRef<() => void>();

  const users: RoleData[] = [
    {role: 'کارشناس امور شهریه', id: 1},
    {role: 'کارشناس امور شهریه', id: 1},
    {role: 'کارشناس امور شهریه', id: 1},
    {role: 'کارشناس امور شهریه', id: 1},
    {role: 'کارشناس امور شهریه', id: 1},
    {role: 'کارشناس امور شهریه', id: 1},
  ];

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <Button
          type="primary"
          className={styles.add_button}
          onClick={() =>
            (dismissDialog.current = showDialog({
              content: <AddRoleDialog onDismissRef={dismissDialog} />,
            }))
          }>
          افزودن نقش
        </Button>
      </div>
      <table className={styles.table_class}>
        <th>ردیف</th>
        <th>عنوان نقش</th>
        <th>عملیات</th>
        {users.map((user, index) => {
          return (
            <tr key={index}>
              <td>{user.id}</td>
              <td>{user.role}</td>
              <td>
                <img
                  src={Assets.Images.EditUsers}
                  alt=""
                  className={styles.edit_icon}
                  onClick={() =>
                    (dismissDialog.current = showDialog({
                      content: <AddRoleDialog onDismissRef={dismissDialog} />,
                    }))
                  }
                />
                <img
                  src={Assets.Images.Delete}
                  alt=""
                  className={styles.delete_icon}
                />
              </td>
            </tr>
          );
        })}
      </table>
    </div>
  );
}
