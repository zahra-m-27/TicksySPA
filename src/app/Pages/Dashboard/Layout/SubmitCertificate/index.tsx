import { Button } from "antd";
import React, { useState } from "react";
import Assets from "../../../../Assets";
import styles from "./styles.module.scss";

interface Request {
  requestStatus: string;
  requestDate: string;
  expirationDate: string;
}

export default function SubmitCertificate() {
  const [Attachment, setAttachment] = useState<File>();
  const [UploadPage, setUploadPage] = useState(true);

  const onSelectFile = (file: File) => {
    setAttachment(file);
  };

  let request: Request = {
    requestStatus: "در حال بررسی",
    requestDate: "1399/11/11",
    expirationDate: "1399/11/11",
  };

  if (UploadPage) {
    return (
      <div className={styles.container_upload}>
        <div className={styles.top_part}>
          <div className={styles.title}>
            <img
              alt=""
              className={styles.title_image}
              src={Assets.Images.FingerPrint}
            />
            احراز هویت
          </div>
        </div>
        <div className={styles.upload_body}>
          <p className={styles.upload_text}>
            لطفا سند احراز هویت خود را بارگذاری کنید
          </p>

          <label htmlFor="attachment" className={styles.attachment_box}>
            <div className={styles.input_box}>
              <img
                alt=""
                src={Assets.Images.Upload}
                className={styles.upload_icon}
              />
              <p className={styles.file_name}>
                {Attachment?.name ?? "انتخاب فایل"}
              </p>
            </div>
          </label>
          <input
            type={"file"}
            id={"attachment"}
            className={styles.input_file}
            onChange={(e) => {
              e.target.files && onSelectFile && onSelectFile(e.target.files[0]);
            }}
          />

          <Button
            type={"primary"}
            className={styles.submit_button}
            onClick={() => {
              setUploadPage(false);
            }}
          >
            ثبت
          </Button>
        </div>
      </div>
    );
  }

  return (
    <div className={styles.container_report}>
      <div className={styles.top_part}>
        <div className={styles.title}>
          <img
            alt=""
            src={Assets.Images.FingerPrint}
            className={styles.report_title_image}
          />
          احراز هویت
        </div>
      </div>
      <div className={styles.report_body}>
        <p className={styles.right_side}>
          <span>سند شما ثبت شد</span>
          <span>لطفا منتظر پاسخ ما باشین</span>
          <span>از صبر و شکیبایی شما </span>
          <span>سپاسگزاریم</span>
        </p>
        <div className={styles.divider} />
        <div className={styles.left_side}>
          <div className={styles.request_data}>
            <div className={styles.request_container}>
              <span className={styles.request}>وضعیت درخواست:</span>
              <div className={styles.request_data_container}>
                {request.requestStatus}
              </div>
            </div>
            <div className={styles.request_container}>
              <span className={styles.request}>تاریخ درخواست:</span>
              <div className={styles.request_data_container}>
                {request.requestDate}
              </div>
            </div>
            <div className={styles.request_container}>
              <span className={styles.request}>تاریخ انقضا:</span>
              <div className={styles.request_data_container}>
                {request.expirationDate}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
