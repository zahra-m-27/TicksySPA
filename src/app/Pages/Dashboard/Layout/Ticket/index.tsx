import API from "../../../../API";
import moment from "jalali-moment";
import Assets from "../../../../Assets";
import styles from "./styles.module.scss";
import { useParams } from "react-router-dom";
import useUser from "../../../../Hooks/useUser";
import React, { useEffect, useState } from "react";
import SEInput from "../../../../Components/SEInput";
import MessageDto from "../../../../API/DTOs/MessageDto";
import TicketDto from "../../../../API/DTOs/TicketDto";
import { message, Spin } from "antd";

export default function Ticket() {
  const { user } = useUser();
  const params = useParams<any>();
  const [Message, setMessage] = useState("");
  const [Loading, setLoading] = useState(false);
  const [Ticket, setTicket] = useState<TicketDto>();
  const [Attachment, setAttachment] = useState<File>();
  const [Messages, setMessages] = useState<MessageDto[]>([]);
  const [MessageHasError, setMessageHasError] = useState(false);

  const getMessages = () => {
    API.Tickets.GetTicketMessages({
      id: params.id,
    })
      .then((response) => {
        setMessages(response);
      })
      .finally(() => setLoading(false));
  };

  useEffect(() => {
    getMessages();
    API.Tickets.GetTicketDetail({ id: params.id }).then((response) => {
      setTicket(response);
    });
  }, []);

  const onSendMessage = () => {
    if (!Message) {
      setMessageHasError(true);
      return;
    }

    let attachments = [];

    if (Attachment) {
      attachments.push(Attachment);
    }

    setLoading(true);
    API.Tickets.CreateTicketMessage({
      id: params.id,
      text: Message,
      attachments: attachments,
    })
      .then(() => {
        getMessages();
      })
      .catch(() => {
        message.error("ارسال پیام مقدور نیست");
      });
  };

  if (!Ticket) {
    return (
      <div className={styles.container}>
        <Spin size="large" />
      </div>
    );
  }

  return (
    <div className={styles.container}>
      <div className={styles.ticket_info_container}>
        <div className={styles.ticket_info_top}>
          <p>عنوان: {Ticket.title}</p>
        </div>
        <div className={styles.ticket_info_bottom}>
          <p>
            تاریخ شروع:{" "}
            {moment
              .utc(Ticket.creation_date)
              .local()
              .locale("fa")
              .format("YYYY/MM/D HH:mm")}
          </p>
          <p>
            آخرین آپدیت:
            {moment
              .utc(Ticket.last_update)
              .local()
              .locale("fa")
              .format("YYYY/MM/D HH:mm")}
          </p>
        </div>
      </div>
      {Ticket.tags && Ticket.tags.length > 0 && (
        <div className={styles.tags_container}>
          <p dir="auto">
            {Ticket.tags
              .split(",")
              .map((tag) => `#${tag}`)
              .join(" ")}
          </p>
          <Assets.SVGs.Hashtag className={styles.hashtag_icon} />
        </div>
      )}

      <div className={styles.chat_container}>
        {Messages.map((message, index) => (
          <div
            key={index}
            className={styles.message}
            style={{
              alignSelf: message.id !== user.id ? "flex-start" : "flex-end",
            }}
          >
            <div
              className={styles.message_user}
              style={{
                justifyContent:
                  message.id !== user.id ? "flex-start" : "flex-end",
                transform: `translate(${
                  7 * (message.id !== user.id ? -1 : 1)
                }px, -7px)`,
              }}
            >
              {message.id !== user.id && (
                <img
                  alt=""
                  src={user.avatar ?? Assets.SVGs.MaleUserSVG}
                  className={styles.message_user_avatar}
                />
              )}
              <p>{message.user.first_name + " " + message.user.last_name}</p>
              {message.id === user.id && (
                <img
                  alt=""
                  src={user.avatar ?? Assets.SVGs.OldMaleUserSVG}
                  className={styles.message_user_avatar}
                />
              )}
            </div>
            <p dir="auto" className={styles.message_content}>
              {message.text}
            </p>
            <div className={styles.message_time}>
              <p>
                {moment
                  .utc(message.date)
                  .local()
                  .locale("fa")
                  .format("YYYY/MM/D")}
              </p>
              <p>
                {moment.utc(message.date).local().locale("fa").format("HH:mm")}
              </p>
            </div>
          </div>
        ))}

        <div className={styles.send_message}>
          {Attachment ? (
            <div className={styles.attachment}>
              <img
                alt="cancel"
                src={Assets.Images.Cancel}
                className={styles.attachment_icon}
                onClick={() => setAttachment(undefined)}
              />
              <p className={styles.name}>{Attachment?.name}</p>
            </div>
          ) : (
            <>
              <label htmlFor="attach_input">
                <Assets.SVGs.Attach className={styles.send_message_icons} />
              </label>
              <input
                type="file"
                id="attach_input"
                onChange={(e) =>
                  e.target.files && setAttachment(e.target.files[0])
                }
                className={styles.attachment_input}
              />
            </>
          )}
          <SEInput
            minLines={2}
            label="پیام شما..."
            onChangeText={setMessage}
            hasError={MessageHasError}
            className={styles.send_message_input}
            innerContainerClassName={styles.send_message_input_innerContainer}
          />
          <div onClick={onSendMessage}>
            {Loading ? (
              <Spin size="large" />
            ) : (
              <Assets.SVGs.PaperPlane className={styles.send_message_icons} />
            )}
          </div>
        </div>
      </div>
    </div>
  );
}
