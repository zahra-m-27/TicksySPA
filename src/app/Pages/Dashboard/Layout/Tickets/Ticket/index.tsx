import API from '../../../../../API';
import moment from 'jalali-moment';
import {message, Spin} from 'antd';
import Assets from '../../../../../Assets';
import styles from './styles.module.scss';
import {useParams} from 'react-router-dom';
import useUser from '../../../../../Hooks/useUser';
import React, {useEffect, useRef, useState} from 'react';
import SEInput from '../../../../../Components/SEInput';
import TicketDto from '../../../../../API/DTOs/TicketDto';
import showDialog from '../../../../../Components/TDialog';
import ForwardTicketDialog from '../../../../../Dialogs/ForwardTicket';
import FroalaEditorView from 'react-froala-wysiwyg/FroalaEditorView';

export default function Ticket() {
  const {user} = useUser();
  const dismissDialog = useRef<() => void>();
  const params = useParams<{ticketId: string}>();
  const [Ticket, setTicket] = useState<TicketDto>();
  const [Message, setMessage] = useState('');
  const [Attachment, setAttachment] = useState<File>();
  const [Loading, setLoading] = useState(false);
  const [MessageHasError, setMessageHasError] = useState(false);

  const getData = () => {
    API.Tickets.GetTicket({id: parseInt(params.ticketId)}).then((response) => {
      setTicket(response);
    });
  };

  useEffect(() => {
    getData();
  }, []);

  const onSendMessage = () => {
    if (!Message) {
      setMessageHasError(true);
      return;
    }

    const attachments = [];

    if (Attachment) {
      attachments.push(Attachment);
    }

    setLoading(true);
    API.Tickets.CreateTicketMessage({
      id: parseInt(params.ticketId),
      text: Message,
      attachments: attachments,
    })
      .then(() => {
        getData();
        setMessage('');
        setAttachment(undefined);
      })
      .catch(() => message.error('ارسال پیام مقدور نیست'))
      .finally(() => setLoading(false));
  };

  if (!Ticket) {
    return (
      <div className={styles.container}>
        <Spin size="large" />
      </div>
    );
  }

  const onForwardingTicket = () => {
    dismissDialog.current = showDialog({
      content: (
        <ForwardTicketDialog
          ticketId={Ticket.id}
          onDismissRef={dismissDialog}
          section={Ticket.section.title}
          otherSections={Ticket.other_sections}
        />
      ),
      style: {
        maxWidth: 'unset',
        overflow: 'hidden',
        borderRadius: 15,
      },
    });
  };

  return (
    <div className={styles.container}>
      <div className={styles.ticket_info_container}>
        <div className={styles.ticket_info_top}>
          <span>{Ticket.title}</span>
          <span dir="auto">عنوان تیکت:</span>
        </div>
        <div className={styles.ticket_info_bottom}>
          <p>
            تاریخ شروع:{' '}
            {moment
              .utc(Ticket.creation_date)
              .local()
              .locale('fa')
              .format('YYYY/MM/D HH:mm')}
          </p>
          <p>
            آخرین آپدیت:
            {moment
              .utc(Ticket.last_update)
              .local()
              .locale('fa')
              .format('YYYY/MM/D HH:mm')}
          </p>
        </div>
      </div>
      {Ticket.tags && Ticket.tags.length > 0 && (
        <div className={styles.tags_container}>
          <p dir="auto">
            {Ticket.tags
              .split(',')
              .map((tag) => `#${tag}`)
              .join(' ')}
          </p>
          <Assets.SVGs.Hashtag className={styles.hashtag_icon} />
        </div>
      )}

      <div className={styles.chat_container}>
        <div className={styles.chat_container_header}>
          <div className={styles.items}>
            <img src={Assets.SVGs.History} alt="history" />
          </div>
          {!!Ticket.other_sections.length && (
            <div
              className={styles.items}
              onClick={onForwardingTicket}
              data-testid="forward-button">
              <img src={Assets.SVGs.ArrowRight} alt="forward" />
            </div>
          )}

          <p dir="auto">مسئول رسیدگی به درخواست شما: </p>
        </div>
        {Ticket.message_set.map((message, index) => {
          const unknownAvatar =
            message.user.id === user.id
              ? Assets.SVGs.MaleUserSVG
              : Assets.SVGs.OldMaleUserSVG;
          return (
            <div
              key={index}
              className={styles.message}
              style={{
                alignSelf:
                  message.user.id !== user.id ? 'flex-start' : 'flex-end',
              }}>
              <div
                className={styles.message_user}
                style={{
                  justifyContent:
                    message.user.id !== user.id ? 'flex-start' : 'flex-end',
                  transform: `translate(${
                    7 * (message.user.id !== user.id ? -1 : 1)
                  }px, -7px)`,
                }}>
                {message.user.id !== user.id && (
                  <img
                    alt=""
                    className={styles.message_user_avatar}
                    src={message.user.avatar ?? unknownAvatar}
                  />
                )}
                <p>{message.user.first_name + ' ' + message.user.last_name}</p>
                {message.user.id === user.id && (
                  <img
                    alt=""
                    className={styles.message_user_avatar}
                    src={message.user.avatar ?? unknownAvatar}
                  />
                )}
              </div>
              <div className={styles.message_content} dir="rtl">
                <FroalaEditorView
                  model={message.text}
                  config={{
                    key: 'yDJ6hT5jU7QUv4Xy4OZh4ABVJRDRNGGUO3ITru7M5a6a21j11e13l12l7w10h7==',
                  }}
                />
              </div>
              <div className={styles.message_time}>
                <p>
                  {moment
                    .utc(message.date)
                    .local()
                    .locale('fa')
                    .format('YYYY/MM/D')}
                </p>
                <p>
                  {moment
                    .utc(message.date)
                    .local()
                    .locale('fa')
                    .format('HH:mm')}
                </p>
              </div>
            </div>
          );
        })}

        <div className={styles.send_message}>
          {Attachment ? (
            <div className={styles.attachment}>
              <img
                alt="cancel"
                src={Assets.Images.Cancel}
                data-testid="cancel_attachment"
                className={styles.attachment_icon}
                onClick={() => setAttachment(undefined)}
              />
              <p className={styles.name}>{Attachment?.name}</p>
            </div>
          ) : (
            <>
              <label htmlFor="attach_input">
                <Assets.SVGs.Attach className={styles.send_message_icons} />
              </label>
              <input
                type="file"
                id="attach_input"
                data-testid="attach_input"
                onChange={(e) =>
                  e.target.files && setAttachment(e.target.files[0])
                }
                className={styles.attachment_input}
              />
            </>
          )}
          <SEInput
            minLines={2}
            content={Message}
            label="پیام شما..."
            onChangeText={setMessage}
            hasError={MessageHasError}
            data-testid="message_input"
            className={styles.send_message_input}
            innerContainerClassName={styles.send_message_input_innerContainer}
          />
          <div onClick={onSendMessage} data-testid="submit">
            {Loading ? (
              <Spin size="large" />
            ) : (
              <Assets.SVGs.PaperPlane className={styles.send_message_icons} />
            )}
          </div>
        </div>
      </div>
    </div>
  );
}
