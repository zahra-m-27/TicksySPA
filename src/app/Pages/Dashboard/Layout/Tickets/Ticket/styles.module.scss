.container {
  display: flex;
  margin-top: 5vh;
  align-items: center;
  flex-direction: column;

  .ticket_info_container {
    width: 93%;
    color: #fff;
    padding: 30px 50px;
    display: flex;
    border-radius: 10px;
    flex-direction: column;
    background-color: #0088e3;

    p {
      margin: 0;
    }

    .ticket_info_top {
      display: flex;
      font-weight: 600;
      text-align: right;
      margin-bottom: 10px;
      justify-content: space-between;
      font-size: clamp(15px, 1.2vw, 22px);
    }

    .ticket_info_bottom {
      display: flex;
      justify-content: space-around;
      font-size: clamp(15px, 1.2vw, 22px);
    }
  }

  .tags_container {
    width: 83%;
    display: flex;
    padding: 27px;
    margin-top: 30px;
    border-radius: 10px;
    justify-content: flex-end;
    background-color: #e0f7ff;

    p {
      color: #9e9797;
      font-size: 18px;
      user-select: none;
      margin: 5px 15px 0;
    }

    .hashtag_icon {
      width: 32px;
      height: 32px;
    }
  }

  .chat_container {
    width: 83%;
    display: flex;
    margin-top: 8vh;
    min-height: 100vh;
    border-radius: 20px;
    margin-bottom: 25px;
    flex-direction: column;
    box-shadow: 0 3px 8px 0 #76787d;
    background-image: url('../../../../../Assets/Images/Files/chat_background.jpg');

    &_header {
      display: flex;
      padding: 15px 25px;
      background: rgba(164, 173, 180, 0.25);

      .items {
        width: 40px;
        height: 40px;
        margin: 0 7px;
        display: grid;
        cursor: pointer;
        border-radius: 50%;
        background: #eff0f4;
        place-content: center;

        &:nth-child(1) img {
          width: 18px;
          height: 18px;
        }

        &:nth-child(2) img {
          width: 25px;
          height: 25px;
        }

        &:nth-child(3) img {
          width: 20px;
          height: 20px;
        }
      }

      p {
        font-weight: 600;
        font-size: clamp(15px, 1vw, 20px);
        margin: auto 0 auto auto;
      }
    }

    .message {
      width: 50%;
      color: white;
      margin: clamp(13px, 4vh, 48px) 60px;
      border-radius: 25px;
      background-color: #2168bd;

      p {
        margin: 0;
      }

      &_user {
        display: flex;
        align-items: center;

        p {
          margin: 0 4rem;
          font-size: 18px;
          font-weight: 600;
        }

        &_avatar {
          width: 75px;
          height: 75px;
          border-radius: 50%;
        }
      }

      &_content {
        margin: 20px !important;
      }

      &_time {
        display: flex;
        user-select: none;
        margin: 0 20px 20px;
        justify-content: space-between;
      }
    }
  }

  .send_message {
    display: flex;
    padding: 0 37px;
    margin-top: auto;
    align-items: center;
    background-color: white;

    &_input {
      margin: 10px 30px;

      &_innerContainer {
        border: hidden !important;
        box-shadow: 0 0 0 0 !important;
      }
    }

    &_icons {
      cursor: pointer;
    }

    .attachment_input {
      display: none;
    }
  }

  .attachment {
    height: 36px;
    display: flex;
    margin: 0 5px;
    padding: 0 10px;
    border-radius: 5px;
    align-items: center;
    transition: all 0.5s;
    background-color: #26a9ad;

    &_icon {
      width: 20px;
      height: 20px;
      cursor: pointer;
      margin-left: 5px;
      transition: all 0.2s;

      &:hover {
        filter: alpha(opacity=0.65);
      }

      &:last-child {
        width: 16px;
        height: 16px;
      }
    }

    &_input {
      display: none;
    }

    .name {
      margin: 0;
      color: #ffffff;
      overflow: hidden;
      user-select: none;
      white-space: nowrap;
    }
  }
}
