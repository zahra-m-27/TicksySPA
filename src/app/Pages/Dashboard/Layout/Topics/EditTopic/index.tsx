import styles from './styles.module.scss';
import {Button, Tabs} from 'antd';
import React, {useRef, useState} from 'react';
import EditTopicDetail from './EditTopicDetail';
import UserManagement from './UserManagement';
import RoleManagement from './RoleManagement';
import showDialog from '../../../../../Components/TDialog';
import ManageTopicRoleDialog from '../../../../../Dialogs/ManageTopicRole';
import TButton from '../../../../../Components/TButton';
import ManageTopicUserDialog from '../../../../../Dialogs/ManageTopicUser';
import {useParams} from 'react-router-dom';

export default function EditTopic() {
  const updateRoles = useRef<() => void>();
  const updateUsers = useRef<() => void>();
  const dismissDialog = useRef<() => void>();
  const params = useParams<{topicId: string}>();
  const [Tab, setTab] = useState('detail');

  let addTitle = '';
  if (Tab === 'roles') {
    addTitle = 'افزودن نقش';
  } else if (Tab === 'users') {
    addTitle = 'افزودن کاربر';
  }

  return (
    <div className={styles.container}>
      <Tabs
        accessKey={Tab}
        onChange={setTab}
        tabBarExtraContent={
          addTitle && {
            right: (
              <TButton
                label={addTitle}
                className={styles.add_button}
                onClick={() => {
                  if (Tab === 'roles')
                    dismissDialog.current = showDialog({
                      content: (
                        <ManageTopicRoleDialog
                          onDismissRef={dismissDialog}
                          onUpdate={updateRoles.current}
                          topicId={parseInt(params.topicId)}
                        />
                      ),
                    });
                  else
                    dismissDialog.current = showDialog({
                      content: (
                        <ManageTopicUserDialog
                          onDismissRef={dismissDialog}
                          onUpdate={updateUsers.current}
                          topicId={parseInt(params.topicId)}
                        />
                      ),
                    });
                }}
              />
            ),
          }
        }>
        <Tabs.TabPane tab="مشخصات" key="detail">
          <EditTopicDetail />
        </Tabs.TabPane>
        <Tabs.TabPane tab="کاربران" key="users">
          <UserManagement onUpdate={updateUsers} />
        </Tabs.TabPane>
        <Tabs.TabPane tab="نقش ها" key="roles">
          <RoleManagement onUpdate={updateRoles} />
        </Tabs.TabPane>
      </Tabs>
    </div>
  );
}
