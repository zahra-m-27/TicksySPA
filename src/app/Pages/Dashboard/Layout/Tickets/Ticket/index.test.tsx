import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import EditTopic from '../../Topics/EditTopic/RoleManagement';
import RoleManagement from '../../Topics/EditTopic/RoleManagement';
import SubmitCertificate from './index';
import Ticket from './index';

const sample = (
  <MemoryRouter>
    <Ticket />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
