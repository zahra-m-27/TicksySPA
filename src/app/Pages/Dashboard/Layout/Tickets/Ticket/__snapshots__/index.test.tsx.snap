// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
